infrastructure/src/main/java/org/corfudb/infrastructure/log/statetransfer/batchprocessor/StateTransferBatchProcessor.java

```diff
@@ -43,54 +48,69 @@
      */
     default TransferBatchResponse writeRecords(ReadBatch readBatch, LogUnitClient logUnitClient,
                                                int writeRetriesAllowed, Duration writeSleepDuration) {
-        List<Long> totalAddressesInRequest = readBatch.getAddresses();
-        Optional<String> destination = readBatch.getDestination();
+        if (readBatch == null || readBatch.getAddresses().isEmpty()) {
+            throw new IllegalArgumentException("The read batch is empty.");
```

Comment:
why we throw an exception if there is an empty list of addresses? Essentially, having an empty request is not an error, it should be ok just do nothing?